### DISCUSSION TITLE
Codestyle VS functionality

### DISCUSSION THREAD

**Discussion by Zyres** [AscEmu/AscEmu#836] [General]
So there is a lot of things we could discuss here, but the most important topic in the last few years was:
**codestyle VS functionality**

Sure we have a codestyle and yes we should use it always, but what to do with a large PR knowing it brings a lot of functionality but does not really fit our codestyle?

Sometimes I review it and add my comments about style to it, sometimes I merge it anyway and fix the style myselfe, but why?
I know that people spend a lot of time to PRs and the moment I feel they get bored and tired of constant changerequests, I merge it and change the style later.

What do you think about it? Should we never ever merge PRs with the wrong style?

**Comment by Adrian-Weihs** [comment]
Does VS have an Auto Formatter like Eclipse? At Work everyone who pushes a new change must have atleast the autoformatter and save actions which will always called at save so the code pushed belongs to our codestyle...
